Mac group tests

Akm0d opened the pull request:
### What does this PR do?
Refactors the PAM auth tests and skips one that doesn't run properly with the threadsafe loader.  
separated from https://github.com/saltstack/salt/pull/54878

dwoz commented:
@Akm0d I don't like the idea of skipping a test because we don't understand why it's failing. Let's take another look at this. Should we close this one while we are investigating more?

Akm0d commented:
> @Akm0d I don't like the idea of skipping a test because we don't understand why it's failing. Let's take another look at this. Should we close this one while we are investigating more?

It shouldn't be closed yet, there may be a bug that the thread safe loader PR doesn't create, but exposes.  I'm looking deeper and getting close, removed the test skipping.

Akm0d committed:
Refactor the mac group tests

 - Simplify setup and teardown logic with user/group.present/absent
 - Added a function for asserting salt-call with no return code
 - Skip tests if required modules and states aren't available

Akm0d commented:
re-run full windows2016
